ynr/apps/candidates/tests/test_models.py

```diff
@@ -27,15 +28,22 @@ def test_get_sopn_view(self):
 
 class TestBallotMethods(TestCase):
     def test_uncontested(self):
-        ballot = self.Ballot(
-            ballot_paper_id="local.sheffield.ecclesall.2021-05-06"
+        contested_ballot = Ballot(
+            ballot_paper_id="local.sheffield.ecclesall.2021-05-06",
+            winner_count=3,
+            candidates_locked=False,
+        )
+        uncontested_ballot = Ballot(
+            ballot_paper_id="local.sheffield.ecclesall.2021-05-06",
+            # winner_count=3,
+            candidates_locked=True,
         )
-        self.assertFalse(ballot.uncontested)
+        # self.assertTrue=(uncontested_ballot.winner_count == Count(self.membership_set))
+        self.assertTrue(uncontested_ballot.uncontested)
 
 
 class TestBallotQuerySetMethods(TestCase):
     def test_uncontested(self):
```

Comment:
Also lost on testing querysets. Advice?

Reply:
Thank you! I did this but I guess it was the mocking that was to blame for failed tests. I'll try with your suggestion of creating objects for both tests.

Reply:
👍 One thing I have thought about using that mixin I suggested - I think I made some recent changes to it that were merged to master that will be missing from this PR. So might be worth rebasing with master so that you have those changes

Reply:
On this last suggested test, in trying to create an election within the scope of that method, I get this error: 
```
      django.db.utils.IntegrityError: duplicate key value violates unique constraint "elections_election_slug_key"
E               DETAIL:  Key (slug)=(sp.2021-05-24) already exists.
```

I don't understand bc I'm not using that election name anywhere else. I've also tried creating an election with a Factory.

Reply:
Yes think its fine to do them in separate tests like this if you prefer!